### DISCUSSION TITLE
Add time tracking

### DISCUSSION THREAD

**Discussion by Zerotask** [community/community#4360] [Projects and Issues]
It would be great to have an option to track time:
- estimated time
- spent time

Here's an example of GitLab's time tracking feature
![image](https://user-images.githubusercontent.com/20150243/123636589-582e1b80-d81d-11eb-8cc0-214afe91b72c.png)

You can also use slash commands to add this information:
- `/estimate 3d 10h`
- `/spend 2h 10m`

This would also allow some statistics for project management.

**Comment by felixfbecker** [comment]
Even just a basic start date field (which can be a prospective start date in the future, or be changed to the actual start date in the present/past once someone starts working on it) and a completion date field  (which can be a due date, or an estimated completion date) would help a lot 🙏🏻

**Reply by redcatbear** [reply]
When you link the issue to a project, you can add custom fields. Unfortunately date fields are not available yet, so you would have to use a string field and post-process it. Not ideal, but a workaround that is already available today.

**Comment by Omzig** [comment]
When moving Devop's closer to Github, the effort fields need to be added:

![image](https://user-images.githubusercontent.com/1987296/148121667-bfbaebaf-b36e-4bac-a3f6-b39e767f5aed.png)

**Comment by redcatbear** [comment]
Time tracking would indeed be very helpful. We do this with a custom field (float) in person-days in GitHub Projects, but that has a couple of downsides:

1. It is not incremental, so team members need to always add the time on top of the existing value
2. It is easy to get this wrong if multiple people add time on the field
3. The field does not have a unit (we just defined it to be person-days)
4. You have to calculate fractions, since there is not input support (it would be more convenient to type: `1w 3d 4h` instead of `8.5`)

The good part about using a project custom field instead of issues is that you can record time in open source projects without disclosing it publicly. Tracking the time someone spent on an issue can be privacy relevant.

**Comment by sovetski** [comment]
I am eagerly awaiting this feature.

This is the only point that blocks me to migrate the management of my projects on GitHub.

It can be done a bit like on Jira (and probably elsewhere it's the same), each person writes the time they spent for example: 1w, 1h or 30m and GitHub saves that as logs and displays a total of time spent, and if we want we can visualize the details.

It is an essential element for each project to determine the time spent

**Reply by data-if** [reply]
when?

**Comment by beatrizmilz** [comment]
Hi! Do we have any update on this request?
I have been using GitHub Projects at work and in community projects, and I miss this feature.

**Comment by gustavokennedy** [comment]
Please @github, it's long past time for this feature in Projects. 🙏🙏

- Total Hours spent per user on each issue.
- Total hours per user monthly/weekly/daily
- Total hours on each issue.

**Comment by dallenbaldwin** [comment]
We just got on the new Issues beta with sub-issues and have been using them extensively. I would really appreciate any time tracking feature to automatically roll-up the time from children into the parents

**Reply by dallenbaldwin** [reply]
It would also be _super_ useful to have this available in insights. Right now, our workaround like others is to have an "Hours" custom field which we have to manually update. We don't use time estimates like others, but use Fibonacci numbers as our estimate. A useful metric for our boss to communicate expectations to stakeholders is the average time spent per Fibonacci number

![image](https://github.com/user-attachments/assets/42aad2d0-417c-4953-a819-b6571f8ec365)
